### DISCUSSION TITLE
Слишком много чекбоксов

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#258] [Как это сделать?] [answered]
Добрый день всем. Прошу помощи у вас. Смысл вот в чем: есть простая таблица и в ней 5 полей чекбоксов (присутствует 5 пользователей, которые будут согласовывать счет). Для каждой роли свое поле с чекбоксом. Вопрос в следующем: как перенести  эти поля с чекбоксами в контекстную панель? Уж слишком большое кол-во дублированных полей с чекбоксами получается.

**Comment by erpic-ru** [comment]
Не до конца понятен кейс
1. Можно каждой роли показать только свой чекбокс
2. В контекстной панели можно разместить кнопки, которые будут ставить нужные чекбоксы и блокироваться на повторное нажатие по чекбоксу true

**Reply by Vadimgrig2012** [reply]
Можно каждой роли показать только свой чекбокс - - не совсем так. Можно что бы к просмотру были доступны все чекбоксы, но изменять каждый могла только определенная роль.

**Comment by erpic-ru** [comment]
Ну ок, менять кнопкам значок на галочку после нажатия, а другим
блокировать. Как засунуть чекбоксы в контекстную панель? скорее всего
никак. Проще держать массив и во временной таблице в iframe редактировать.
Недавно обсуждали

ср, 2 нояб. 2022 г. в 18:58, Vadimgrig2012 ***@***.***>:

> Можно каждой роли показать только свой чекбокс - - не совсем так. Можно
> что бы к просмотру были доступны все чекбоксы, но изменять каждый могла
> только определенная роль.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/258#discussioncomment-4038771>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AV3NMAUSBG4LDFB5SXF6SOTWGKFRXANCNFSM6AAAAAARVGW5WU>
> .
> You are receiving this because you commented.Message ID:
> <totumonline/totum-ru-issues-and-discussions/repo-discussions/258/comments/4038771
> @github.com>
>


-- 
Timur

**Comment by ttm-alexeypolunin** [comment]
Есть у меня хорошая задумка...

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Могу предложить такой вариант:

<img width="845" alt="Снимок экрана 2022-11-08 в 15 02 29" src="https://user-images.githubusercontent.com/55755565/200559109-c22c3a85-7360-425a-9cc6-b9e9a6d6d65f.png">

Мы скрываем поля чекбоксов, а оставляем одно с ручным Селектом в котором отображаем, что выбрано.

Селект я сформировал с основаниями как у `name` полей чекбоксов, чтоб не запутаться. Тип `json` в селекте — `object`:

<img width="733" alt="Снимок экрана 2022-11-08 в 15 04 01" src="https://user-images.githubusercontent.com/55755565/200559588-ef509567-8f0a-4d24-8b0f-2bc5521c3fd2.png">

<img width="329" alt="Снимок экрана 2022-11-08 в 15 04 32" src="https://user-images.githubusercontent.com/55755565/200559606-8ecd33c6-0c3f-4482-a15d-5f5cc0426189.png">

И теперь кодом считаем значения в этом Селекте:

```
= : rowKeys(row: $filter)

filter: listFilter(list: $row; key: "value" = true)
	row: rowCreate(field: "ch1" = #ch1; field: "ch2" = #ch2; field: "ch3" = #ch3)

```

Здесь очень хочется взять `selectRow` по этой же строке, но это **не сработает!** Расчетные значения будут опаздывать на один шаг, тк `select`-функции берут в кодах значения из БД в то время как Код исполняется до того, как была сделана запись в БД. Поэтому здесь нужен `rowCreate` и обращение через `#`.

В кнопке `Apply` Код действия, который ставит через `set` галочку в нужную колонку в зависимости от роли.


**PS:** Через некоторое время в PRO все это можно будет сделать без технических полей с чекбоксами, так-как у префильтров появится возможность отрицания.
